### ISSUE TITLE
pytest fixture setup reported as additional test

### ISSUE THREAD

**Issue by westshawn** [JetBrains/teamcity-messages#91]
The fixture setup is reported as a separate (additional) test.
When the base test is expanded in the build log, the fixture setup line also appears, indented under the base test, but expanding does nothing except display "Loading..." indefinitely.

The fixture setup should NOT be registered as a separate test.
The fixture setup should be expandable under the base test.

Thanks for taking a look,
-Shawn

platform win32 -- Python 2.7.3, pytest-2.8.5, py-1.4.31, pluggy-0.3.1
TeamCity 9.1.5 (build 37377), teamcity-messages 1.17

build log collapsed:
```
[13:03:23][Step 1/13] Test_pytest.TestClass.test_brie
[13:03:23][Step 1/13] 
[13:03:23][Step 1/13] Test_pytest.TestClass.test_brie_setup
[13:03:23][Step 1/13] 
[13:03:23][Step 1/13] 
[13:03:23][Step 1/13] .
```
build log expanded:
```
[13:03:23][Step 1/13] Test_pytest.TestClass.test_brie
[13:03:23]   [Test_pytest.TestClass.test_brie] 
[13:03:23]   [Test_pytest.TestClass.test_brie] Test_pytest.TestClass.test_brie_setup
 Loading...
[13:03:23]   [Test_pytest.TestClass.test_brie] 
[13:03:23]   [Test_pytest.TestClass.test_brie] 
[13:03:23]   [Test_pytest.TestClass.test_brie] .
[13:03:23]
			 [Test_pytest.TestClass.test_brie] [Test Output]
			 in test_brie()
[13:03:23]   [Test_pytest.TestClass.test_brie] 
[13:03:23][Step 1/13] 
[13:03:23][Step 1/13] Test_pytest.TestClass.test_brie_setup
[13:03:23]   [Test_pytest.TestClass.test_brie_setup] 
[13:03:23]
			 [Test_pytest.TestClass.test_brie_setup] [Test Output]

			 [setup] cheese_db, connect to db
[13:03:23]   [Test_pytest.TestClass.test_brie_setup] 
[13:03:23][Step 1/13] 
[13:03:23][Step 1/13] 
[13:03:23][Step 1/13] .
```
test code
```
class TestClass:
	@pytest.fixture()
	def cheese_db(self, request):
		print('\n[setup] cheese_db, connect to db')
		# code to connect to your db
		a_dictionary_for_now = {'Brie': 'No.', 'Camenbert': 'We do!'}

		def fin():
			print('\n[teardown] cheese_db finalizer, disconnect from db')
			request.addfinalizer(fin)

		return a_dictionary_for_now

	def test_brie(self, cheese_db):
		print('in test_brie()')
		assert cheese_db['Brie'] == 'No.'
```

**Comment by westshawn** [comment]
Note: When "[Test_pytest.TestClass.test_brie] Test_pytest.TestClass.test_brie_setup" finishes loading, it it is a child (indented) of the base test [Test_pytest.TestClass.test_brie] and contains the expected output from the setup function.

When the duplicate test [Test_pytest.TestClass.test_brie_setup] is clicked, it contains no data, but toggles the child of the base test in the log above ([Test_pytest.TestClass.test_brie] Test_pytest.TestClass.test_brie_setup)

**Comment by shalupov** [comment]
I still need to report setup fixture output somewhere.

I'll try to use message blocks https://confluence.jetbrains.com/display/TCD9/Build+Script+Interaction+with+TeamCity#BuildScriptInteractionwithTeamCity-BlocksofServiceMessages

That way test output will be in the build log, expandable/collapsible, but not in the tests list.

Failed setup fixtures will be still reported as failed tests

**Comment by westshawn** [comment]
@shalupov, 
Thanks for your work on this!

Message blocks is a great idea.

Is it possible for a failed setup fixture to force all dependent tests to report as failed, but not create a separate (additional) test for the fixture itself?

If the fixture reports as a separate test, then the test count changes, which is misleading.

**Comment by shalupov** [comment]
> force all dependent tests to report as failed
We're reporting what pytest is reporting to test listener, so it's somewhat limited

**Comment by westshawn** [comment]
I see now- as expected, pytest is identifying and reporting only the defined tests (not setup).

If I have two tests that depend on fixture setup, then pytest reports two tests.
When there is an error in fixture setup, pytest reports 2 tests failed:
```
python -m py.test -r a Test_pytest.py
...
=========================== short test summary info ===========================
ERROR Test_pytest.py::TestClass::()::test_brie[cheese_db]
ERROR Test_pytest.py::TestClass::()::test_cheddar[cheese_db]
================= 2 error in 1.79 seconds ==================
```

But, TeamCity captures the following tests:
* 3 tests when fixture setup is successful
  * TestClass.test_brie[cheese_db]
  * TestClass.test_brie[cheese_db]_setup
  * TestClass.test_cheddar[cheese_db]
* 4 tests when there is an error in fixture setup
  * TestClass.test_brie[cheese_db]
  * TestClass.test_brie[cheese_db]_setup
  * TestClass.test_cheddar[cheese_db]
  * TestClass.test_cheddar[cheese_db]_setup

This request is to make TeamCity report the same number of tests that pytest reports.
I like the idea of capturing fixture setup output as a sub-block in the build log under each dependent test without reporting "*_setup" as distinct test(s).

**Comment by joshrlesch** [comment]
Any progress on this?

**Comment by shalupov** [comment]
Sorry, not yet. I thin I'll return to the project in a couple of weeks

**Comment by shalupov** [comment]
Finally fixed

**Comment by joshrlesch** [comment]
Sweet, awesome news! Good work. Any idea when a new version will be released so we can take advantage of this change?

**Comment by shalupov** [comment]
released 1.18
